pontoon/base/models.py

```diff
@@ -2026,12 +2026,28 @@ def for_project_locale(self, project, locale, paths=None, statuses=None,
         if search:
             # https://docs.djangoproject.com/en/dev/topics/db/queries/#spanning-multi-valued-relationships
             search_query = (search, locale.db_collation)
-            entities = (
-                Entity.objects.filter(
-                    Q(translation__string__icontains_collate=search_query, translation__locale=locale) | Q(string__icontains=search) | Q(string_plural__icontains=search) | Q(comment__icontains=search) | Q(key__icontains=search),
-                    pk__in=entities
+            translation_matches = (
+                entities.filter(
+                    translation__string__icontains_collate=search_query,
+                    translation__locale=locale,
                 )
                 .distinct()
+                .values_list('id')
+            )
+            entity_matches = (
+                entities.filter(
+                    Q(string__icontains=search) |
+                    Q(string_plural__icontains=search) |
+                    Q(comment__icontains=search) |
+                    Q(key__icontains=search)
+                )
+                .distinct()
+                .values_list('id')
+            )
+            entities = Entity.objects.filter(
+                pk__in=set(
+                    e[0] for e in list(translation_matches) + list(entity_matches)
+                )
```

Comment:
...and then you can just `set(list(translation_matches) + list(entity_matches))`.